### ISSUE TITLE
The weather model hash should be updated to the latest main branch

### ISSUE THREAD

**Issue by mkavulich** [ufs-community/ufs-srweather-app#1049]
## Description
The hash for ufs-weather-model in SRW has fallen behind the latest development by several months, and there are some major incompatibilities with the current workflow that need to be solved. This is a problem for a project I'm working with (UFS_FIRE) which has capabilities that are even with the latest weather model development, and so will not work in the latest SRW workflow.

## Solution
The weather model hash should be updated to the latest hash ([8518c2c](https://github.com/ufs-community/ufs-weather-model/commit/8518c2c59fd92ec8a5982997d66612992b577dda) or later) of ufs-weather-model. This will require at least three additional fixes in order to work correctly:

 - Ozone climatology file symlink logic needs to be updated. This is due to a reorganization of the ozone physics code, where this parameterization is now only controlled via namelist option rather than via both namelist and SDF. I should be able to make these changes.
 - Some file names in the weather model have changed. I should be able to make these changes.
 - Spack-stack modules need to be updated to version 1.5.1. Since it is used by the weather model I assume this should be as simple as changing our build module files, but **I will need EPIC or other outside help in order to ensure this is working on machines I do not have access to**.

## Requirements**
The SRW should build and run with the latest hash ([8518c2c](https://github.com/ufs-community/ufs-weather-model/commit/8518c2c59fd92ec8a5982997d66612992b577dda) or later) of ufs-weather-model

## Acceptance Criteria (Definition of Done)
The SRW should build and run with the latest hash ([8518c2c](https://github.com/ufs-community/ufs-weather-model/commit/8518c2c59fd92ec8a5982997d66612992b577dda) or later) of ufs-weather-model

**Comment by MichaelLueken** [comment]
@mkavulich - I have started updating the UFS_UTILS, UPP, and ufs-weather-model hashes in my [feature/hash_update](https://github.com/MichaelLueken/ufs-srweather-app/tree/feature/hash_update) branch.  I have the majority of the necessary modification made and the fundamental and comprehensive tests have successfully passed.

I will need to update the ufs-weather-model hash to [8518c2c](https://github.com/ufs-community/ufs-weather-model/commit/8518c2c59fd92ec8a5982997d66612992b577dda) and apply the updates for spack-stack v.1.5.1 and retest, but that should be easier than the majority of the other changes that I have already worked through in this branch.

**Comment by MichaelLueken** [comment]
@mkavulich - I have updated my [feature/hash_update](https://github.com/MichaelLueken/ufs-srweather-app/tree/feature/hash_update) branch to [24d83ca](https://github.com/MichaelLueken/ufs-srweather-app/commit/24d83cac1dad340424ef70bb8f28e3b21349d79c).  This updates the ufs-weather-model hash to [8518c2c](https://github.com/ufs-community/ufs-weather-model/commit/8518c2c59fd92ec8a5982997d66612992b577dda), the UPP hash to [945cb2c](https://github.com/NOAA-EMC/UPP/commit/945cb2cef5e8bd5949afd4f0fc35c4fb6e95a1bf), and updates the spack-stack version to 1.5.1.  These have been tested on Hercules, where the fundamental tests have passed.

I will need to run the comprehensive tests, and also ensure that the SRW builds on other machines, but this is a good start.

**Comment by mkavulich** [comment]
@MichaelLueken Thanks for working on this! It looks like your changes are close to mine that I've already tested. My only recommendation would be to remove the `ush/set_ozone_param.py` file entirely since it's now just a bunch of additional unneeded logic; see my commit here: https://github.com/ufs-community/ufs-srweather-app/commit/c9217d7a1f2caa434b3d9672fb3504766b92b46d

**Comment by MichaelLueken** [comment]
Thans for recommendation on removing the `ush/set_ozone_param.py` file, @mkavulich!  I have applied this change in my working copy and am testing it now.

I've also pushed a small update for Jet.  The `modulefiles/build_jet_intel.lua` file has `load("prod-util/1.2.2")`.  The updated spack-stack doesn't have prod-util, but this is a remnant from when RRFS components were being merged to the SRW.  This module is required for GSI, but since GSI isn't in the SRW App, it is okay to remove.
